history.go

```diff
@@ -0,0 +1,88 @@
+package merkletree
+
+import (
+	"errors"
+
+	"github.com/coniks-sys/libmerkleprefixtree-go/crypto"
+)
+
+var (
+	ErrBadEpoch           = errors.New("[merkletree] Next epoch's STR has bad epoch")
+	ErrSTRNotFound        = errors.New("[merkletree] STR not found")
+	ErrMalformedHashChain = errors.New("[merkletree] Malformed hash chain")
+)
+
+type History struct {
+	currentSTR    *SignedTreeRoot
+	hashChain     map[int64]*SignedTreeRoot
+	epochs        []int64 // slice of epochs in hashChain
+	epochInterval int64
+}
+
+func NewHistory(m *MerkleTree, key crypto.SigningKey, startEp, epInterval, size int64) *History {
```

Comment:
I think, it could make sense to add a comment, what `size` actually is or give a slightly more specific name.